rest/api_test.go

```diff
@@ -5106,3 +5106,76 @@ func TestImportOnWriteMigration(t *testing.T) {
 	response = rt.SendAdminRequest("PUT", "/db/doc1?rev=2-44ad6f128a2b1f75d0d0bb49b1fc0019", `{"value":"newer"}`)
 	assertStatus(t, response, http.StatusCreated)
 }
+
+func TestAttachmentContentType(t *testing.T) {
+	rt := NewRestTester(t, nil)
+	defer rt.Close()
+
+	type attTest struct {
+		setContentType                bool
+		putContentType                string
+		expectedContentDispositionSet bool
+	}
+
+	tests := []attTest{
+		{
+			setContentType:                true,
+			putContentType:                "text/html",
+			expectedContentDispositionSet: true,
+		},
```

Comment:
Might be worth adding a test case with a parameter

```suggestion
		{
			setContentType:                true,
			putContentType:                "text/html",
			expectedContentDispositionSet: true,
		},
		{
			setContentType:                true,
			putContentType:                "text/html; charset=utf-8",
			expectedContentDispositionSet: true,
		},
```